samples/lib/thrift/SSLsocket/hello_client/src/main.cpp

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2022 Meta
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifdef __ZEPHYR__
+#include <zephyr.h>
+#endif
+
+#include <cstdlib>
+#include <iostream>
+
+#include <fcntl.h>
+
+#include <thrift/protocol/TBinaryProtocol.h>
+#include <thrift/transport/TBufferTransports.h>
+#include <thrift/transport/TSSLSocket.h>
+
+#include "Hello.h"
+
+using namespace apache::thrift;
+using namespace apache::thrift::protocol;
+using namespace apache::thrift::transport;
+
+#ifdef __ZEPHYR__
+int main(void)
+#else
+int main(int argc, char** argv)
+#endif
+{
+  std::string my_addr;
+
+#ifdef __ZEPHYR__
+  my_addr = CONFIG_NET_CONFIG_PEER_IPV4_ADDR;
+#else
+  if (argc >= 2) {
+    my_addr = std::string(argv[1]);
+  } else {
+    my_addr = "192.0.2.1";
+  }
+#endif
+
+  int port = 4242;
+
+  std::shared_ptr<TSSLSocketFactory> socketFactory = std::make_shared<TSSLSocketFactory>();
+  socketFactory->authenticate(true);
+#ifdef __ZEPHYR__
```

Comment:
I think this logic is wrong - this file is always the client whether or not it is being built for Zephyr or the host.